guillotina/contentnegotiation.py

```diff
@@ -1,844 +1,34 @@
-                continue
-            elif len(allowable) == 1:
-                # we found exactly one match, so this is our content type to use
-                return allowable[0]
-            else:
-                # we found multiple supported content types at this q value, so now we need
-                # to choose the server's preference
-                for i in range(len(server)):
-                    # iterate through the server explicitly by numerical position
-                    if server[i] in allowable:
-                        # when we find our first content type in the allowable list, it is
-                        # the highest ranked server content type that is allowable, so
-                        # this is our type
-                        return server[i]
-
-        # we've got to here without returning anything, which means that the client and
-        # server can't come to an agreement on what content type they want and can deliver.
-        # There's nothing more we can do!
-        return None
-
-
-@implementer(IContentNegotiation)
-class ContentNegotiatorUtility(object):
-
-    def __init__(self, header, enabled_values):
-        if header == 'content_type':
-            server = [AcceptParameters(content_type=ContentType(x)) for x in enabled_values]
-            self.cn = ContentNegotiator(acceptable=server)
-        if header == 'language':
-            server = [AcceptParameters(language=Language(x)) for x in enabled_values]
-            self.cn = ContentNegotiator(acceptable=server)
-
-    def negotiate(self, **kwargs):
-        return self.cn.negotiate(**kwargs)
-
-
-def content_type_negotiation(request, resource, view):
+def get_ordered_content_type_value_list(value):
+    '''
+    ca-ES,ca;q=0.9,en;q=0.8,es;q=0.7
+    '''
+    result = []
+    for ct in value.lower().split(','):
+        weight = 1.0
+        ct_name = ct.split(';')[0]
+        for param in ct.split(';')[1:]:
+            if param.startswith('q='):
+                try:
+                    weight = float(param[len('q='):])
+                except ValueError:
+                    pass
+        result.append((weight, ct_name))
+    return [i[1] for i in sorted(result, reverse=True)]
+
+
+def get_acceptable_content_types(request):
     # We need to check for the language
 
-    accept = None
-
     if 'ACCEPT' in request.headers:
-        accept = request.headers['ACCEPT']
-
-    if IDownloadView.providedBy(view) or accept in (None, '*/*'):
-        # if download view, we want to render raw immediately
-        # or if no or */* accept header provided
-        return IRendererFormatRaw
-
-    np = get_utility(IContentNegotiation, 'content_type')
-    ap = np.negotiate(accept=accept)
-    # We need to check for the accept
-    if str(ap.content_type) in app_settings['renderers']:
-        return app_settings['renderers'][str(ap.content_type)]
+        return get_ordered_content_type_value_list(request.headers['ACCEPT'])
```

Comment:
We are losing the difference between type and subtype. Maybe we need two different parsers (like the code we are removing)